crates/transparency/src/map/map.rs

```diff
@@ -261,14 +261,9 @@ where
 /// // Empty tree:
 /// H(0b00)
 /// ```
-pub(crate) fn hash_branch<D>(lhs: Option<Hash<D>>, rhs: Option<Hash<D>>) -> Hash<D>
+pub(crate) fn hash_branch<D>(lhs: &Hash<D>, rhs: &Hash<D>) -> Hash<D>
 where
     D: SupportedDigest,
 {
-    match (lhs, rhs) {
-        (Some(left), Some(right)) => Hash::of(&(0b11, left, right)),
-        (Some(left), None) => Hash::of(&(0b10, left)),
-        (None, Some(right)) => Hash::of(&(0b01, right)),
-        (None, None) => Hash::of(0b00u8),
-    }
+    Hash::of(&(lhs, rhs))
```

Comment:
Lann pointed out earlier that we should still use a prefix for leaf and branch so that collisions aren't possible even if the client does something a little wrong.
```suggestion
    Hash::of(&(0b1, lhs, rhs))
```